ethexe/tx-pool/src/tests.rs

```diff
@@ -0,0 +1,112 @@
+// This file is part of Gear.
+//
+// Copyright(C) 2025 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Testing module for the tx pool.
+//!
+//! Test here mainly focus on:
+//! - the overall logic of the tx pool to work as expected
+
+use crate::{
+    OffchainTransaction, RawOffchainTransaction, SignedOffchainTransaction, TxPoolService,
+};
+use ethexe_db::{BlockHeader, BlockMetaStorage, Database, MemDb};
+use ethexe_signer::{PrivateKey, Signer, ToDigest};
+use gprimitives::{H160, H256};
+use parity_scale_codec::Encode;
+use std::str::FromStr;
+
+pub(crate) fn generate_signed_ethexe_tx(reference_block_hash: H256) -> SignedOffchainTransaction {
+    let signer = Signer::tmp();
+    let public_key = signer
+        .add_key(
+            PrivateKey::from_str(
+                "4c0883a69102937d6231471b5dbb6204fe51296170827936ea5cce4b76994b0f",
+            )
+            .expect("invalid private key"),
+        )
+        .expect("key addition failed");
+
+    let transaction = OffchainTransaction {
+        raw: RawOffchainTransaction::SendMessage {
+            program_id: H160::random(),
+            payload: vec![],
+        },
+        reference_block: reference_block_hash,
+    };
+    let signature = signer
+        .sign_digest(public_key, transaction.encode().to_digest())
+        .expect("signing failed");
+
+    SignedOffchainTransaction {
+        transaction,
+        signature: signature.encode(),
+    }
+}
+
+pub(crate) fn new_block(parent_hash: Option<H256>) -> (H256, BlockHeader) {
+    let block_hash = H256::random();
+    let header = BlockHeader {
+        height: 0,
+        timestamp: 0,
```

Comment:
I'd avoid potential invalidations of test env caused by incorrect height and timestamp

Reply:
done